app/controllers/application_controller.rb

```diff
@@ -1,14 +1,26 @@
 class ApplicationController < ActionController::Base # :nodoc:
+  respond_to :json
   protect_from_forgery with: :null_session
 
+  include Pundit
+
   before_action :configure_permited_parameters, if: :devise_controller?
 
-  acts_as_token_authentication_handler_for User
+  respond_to :json
+
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
```

Comment:
do something similar to this for when a find method throws a not found error, with this rescue_from ActiveRecord::RecordNotFound, with: :record_not_found